orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy

```diff
@@ -16,19 +16,135 @@
 package com.netflix.spinnaker.orca.pipeline.util
 
 import com.fasterxml.jackson.databind.ObjectMapper
-import com.netflix.spinnaker.orca.pipeline.model.Execution
+import com.netflix.spinnaker.orca.pipeline.StageDefinitionBuilder
 import com.netflix.spinnaker.orca.pipeline.model.Pipeline
 import com.netflix.spinnaker.orca.pipeline.model.PipelineStage
 import com.netflix.spinnaker.orca.pipeline.model.Stage
+import com.netflix.spinnaker.orca.pipeline.model.TestPipelineStage
 import org.springframework.beans.factory.annotation.Autowired
 import spock.lang.Specification
 import spock.lang.Unroll
 
+import java.util.function.BiFunction
+
 class PackageInfoSpec extends Specification {
 
   @Autowired
   ObjectMapper mapper
 
+  def "If no artifacts in current build info or ancestor stages, code should execute as normal"() {
+    final Pipeline pipeline = Pipeline.builder().withTrigger(["buildInfo": ["artifacts": [["fileName": "testFileName"]]]]).build()
+    final Stage quipStage = new PipelineStage()
```

Comment:
Believe there's a constructor on `PipelineStage` that would let you pass in the pipeline and context.